lib/netext/httpext/request.go

```diff
@@ -287,6 +288,11 @@ func MakeRequest(ctx context.Context, preq *ParsedHTTPRequest) (*Response, error
 	mreq := preq.Req.WithContext(ctx)
 	res, resErr := client.Do(mreq)
 
+	if resErr == nil && res.StatusCode == http.StatusSwitchingProtocols {
```

Comment:
Maybe use `if resErr != nil && res.StatusCode == http.StatusSwitchingProtocols`

Reply:
I guess you mean "res != nil" ? 
From the [docs](https://golang.org/pkg/net/http/#Client.Do):
>  On error, any Response can be ignored. A non-nil Response with a non-nil error only occurs when CheckRedirect fails, and even then the returned Response.Body is already closed. 

And I would say in that case the above code is okay I guess it's more accurate to check  if `res` is not nil as maybe there will be some rare case where this could happen ?!? But I doubt it

Reply:
Yeah, I considered checking `res != nil` here, but ended up leaving it. @MStoykov is right, it would be more accurate and safer to check for that instead. I'll change it, thanks.

Reply:
Yes, my typo.

Actually, `reErr == nil` will imply `res != nil`, the 2 checks are equivalent, but using `res != nil` looks more relevant to next condition.